buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -621,7 +621,7 @@ private boolean allocateWithoutLock(int size, int maxCapacity, AdaptiveByteBuf b
         }
 
         private boolean allocate(int size, int maxCapacity, AdaptiveByteBuf buf) {
-            recordAllocationSize(buf.length);
+            recordAllocationSize(size);
```

Comment:
I think we should better add some javadoc to this method, even it's private.

Reply:
Nope, `size` is what is requested and is sampled in the histogram (per size buckets) - with the purpose of computing some percentiles out of it, on rotation